### DISCUSSION TITLE
Using derivatives with random smooths by individual

### DISCUSSION THREAD

**Discussion by TSeuH** [gavinsimpson/gratia#242] [Q&A] [answered]
Hello,

I have been trying to get the first derivatives for each individual fit from the following model:

gam.smooth <- gam(TV ~ groupID +
                    s(time, by = groupID) +
                    s(time, animalID, bs = "fs", m = 1),
                  data = dataTV, method = "REML")

fdSmooth <- derivatives(gam.smooth, n = 1000, term = "s(time)", partial_match = TRUE)

I see a column "fs_var" in fdSmooth, and I was assuming it would contain "animalID", just like by_var contains "groupID". I thought I could get the first derivative for each animalID, but with this code, I only get the first derivative for each groupID, regardless of the animalID.

Would it be possible to have the derivatives by animalID without changing the GAM?

Thank you.

**Comment by gavinsimpson** [comment]
I will take a closer look later, but initial thoughts are that you need `term = "s(time,animalID)"` as that's how {mgcv} refers to the smooth and the pattern `"s(time)"` doesn't match because the closing parenthesis is not after `time` for the `fs` smooth

**Comment by gavinsimpson** [comment]
I'll also add that I doubt the derivative will be that informative if you set `m=1`. For more reliable estimation of derivatives you need `m > d` where `d` is the order of derivative you wish to estimate.

**Comment by TSeuH** [comment]
Thank you for your answer!
Unfortunately, I get an error message when I try your suggestion:

fdSmooth <- derivatives(gam.smooth, n = 1000, term = "s(time, animalID)", partial_match = TRUE)
Erreur : Failed to match any smooths in model.
Try with 'partial_match = TRUE'?

**Comment by TSeuH** [comment]
I have just noticed that if I remove the space between time and animalID, I get a different error message:

derivatives(gam.smooth, n = 1000, term = "s(time,animalID)", partial_match = TRUE)
Error in `str_detect()`:
! Can't recycle `string` (size 2) to match `pattern` (size 3).
Run `rlang::last_trace()` to see where the error occurred.

**Reply by gavinsimpson** [reply]
Can you try that last one without the `partial_match = TRUE`? You don't need that when you are specifying the full name of the smooth.

**Reply by TSeuH** [reply]
Thank you for the explanations. Unfortunately, it gives the same error on str_detect().

**Comment by gavinsimpson** [comment, marked as answer]
I can't reproduce the issue with an example that is similar to the one you describe: see the example I posted in #243

I may have inadvertently fixed this. I'm going to close #243 now. If your problem persists with the CRAN version of gratia, please follow up here or in #243, but I'm going to need a reproducible example I think.
